app/models/policy.rb

```diff
@@ -49,10 +49,15 @@ def fill_from(profile:)
   def update_hosts(new_host_ids)
     return unless new_host_ids
 
-    removed = policy_hosts.where.not(host_id: new_host_ids).destroy_all
-    imported = PolicyHost.import((new_host_ids - host_ids).map do |host_id|
-      { host_id: host_id, policy_id: id }
-    end)
+    removed = imported = nil
+
+    PolicyHost.transaction do
```

Comment:
This was designed for the caller to wrap in a transaction see https://github.com/RedHatInsights/compliance-backend/blob/master/app/controllers/v1/profiles_controller.rb#L45. The GraphQL mutation could do the same, or we should remove the transation within our REST controller, but it should be consistent.

Reply:
So I can just rely on the fact that there's a `transaction` around?

Reply:
Yes, we can rely on the calling code to wrap this in a transaction. We should revisit the graphql mutation: https://github.com/RedHatInsights/compliance-backend/pull/768/files#r590519204